client/signup/steps/p2-complete-profile/index.jsx

```diff
@@ -0,0 +1,163 @@
+import { Button } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useTranslate } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import EditGravatar from 'calypso/blocks/edit-gravatar';
+import FormInputValidation from 'calypso/components/forms/form-input-validation';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import P2StepWrapper from 'calypso/signup/p2-step-wrapper';
+import { saveUserSettings } from 'calypso/state/user-settings/actions';
+import {
+	hasUserSettingsRequestFailed,
+	isUpdatingUserSettings,
+} from 'calypso/state/user-settings/selectors';
+
+import './style.scss';
+
+function P2CompleteProfile( {
+	flowName,
+	stepName,
+	positionInFlow,
+	submitSignupStep,
+	goToNextStep,
+} ) {
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const [ formFullName, setFormFullName ] = useState( '' );
+	const [ formErrors, setFormErrors ] = useState( {} );
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const updatingUserSettings = useSelector( isUpdatingUserSettings );
+	const userSettingsRequestFailed = useSelector( hasUserSettingsRequestFailed );
+
+	useEffect( () => {
+		if ( isSubmitting && ! updatingUserSettings ) {
+			setIsSubmitting( false );
+
+			if ( ! userSettingsRequestFailed ) {
+				recordTracksEvent( 'calypso_signup_p2_complete_profile_step_submit' );
+
+				const stepData = {
+					stepName: stepName,
+					formFullName,
+				};
+
+				submitSignupStep( stepData );
+
+				goToNextStep();
+			}
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ isSubmitting, updatingUserSettings, userSettingsRequestFailed ] );
```

Comment:
Any particular reason why we're not listing all dependencies of this hook here?

Reply:
I was thinking to include only those which could actually change and be important to react to their change. The others are just functions and stuff. I'll include them all.